include/extractor/io.hpp

```diff
@@ -96,6 +99,51 @@ inline void write(const boost::filesystem::path &path, const SegmentDataContaine
     writer.WriteFrom(segment_data.rev_durations);
     writer.WriteFrom(segment_data.datasources);
 }
+
+// read/write for conditional turn restrictions file
+inline void read(const boost::filesystem::path &path, std::vector<InputRestrictionContainer> &restrictions)
+{
+    const auto fingerprint = storage::io::FileReader::VerifyFingerprint;
+    storage::io::FileReader reader{path, fingerprint};
+
+    auto num_indices = reader.ReadElementCount32();
```

Comment:
I would make all counts 64 bits in your new code already; see also https://github.com/Project-OSRM/osrm-backend/pull/3848